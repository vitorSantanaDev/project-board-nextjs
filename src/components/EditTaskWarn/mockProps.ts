export const handleClickCancelEdit = jest.fn()
