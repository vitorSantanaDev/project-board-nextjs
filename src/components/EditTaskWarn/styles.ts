import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  margin-bottom: 2rem;
  transition: ease-in-out 0.3s;
`

export const ButtonCloseWarn = styled.button`
  border: none;
  cursor: pointer;
  padding: 0.75rem;
  margin-left: 8px;
  margin-right: 12px;
  background-color: transparent;
`

export const MessageWarn = styled.span`
  color: #ffb800;
  font-size: 1.5rem;
`
