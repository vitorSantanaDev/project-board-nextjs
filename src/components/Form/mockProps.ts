export const formProps = {
  handleAddTaskSubmit: jest.fn(),
  handleChangeInput: jest.fn(),
  input: ''
}
