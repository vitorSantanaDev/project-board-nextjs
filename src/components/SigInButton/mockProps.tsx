export const mockProps = {
  session: true,
  onClick: jest.fn()
}
