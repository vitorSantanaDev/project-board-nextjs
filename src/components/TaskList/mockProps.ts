export const tasksProps = [
  {
    id: '3896433',
    userId: '8458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '39033',
    userId: '8456458',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '38933',
    userId: '865658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '3366553',
    userId: '85654658',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  },
  {
    id: '337763',
    userId: '88',
    name: 'Name the User',
    tarefa: 'Name Task',
    created: new Date(),
    createdFormated: 'Date of Created'
  }
]
