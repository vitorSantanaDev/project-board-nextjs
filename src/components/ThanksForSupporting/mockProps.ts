export const mockProps = {
  lastDonateUser: `Sua última doação foi a ${new Date().toLocaleDateString()}`
}
