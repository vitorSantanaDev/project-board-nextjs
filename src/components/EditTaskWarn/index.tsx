import React from 'react'

import { FiX } from 'react-icons/fi'

import * as S from './styles'

interface EditTaskWarnProps {
  handleClickCancelEdit: () => void
}

const EditTaskWarn: React.FC<EditTaskWarnProps> = ({
  handleClickCancelEdit
}) => {
  return (
    <S.Wrapper>
      <S.ButtonCloseWarn onClick={handleClickCancelEdit}>
        <FiX size={30} color="#FF3636" />
      </S.ButtonCloseWarn>
      <S.MessageWarn>Você está editando uma tarefa!</S.MessageWarn>
    </S.Wrapper>
  )
}

export default EditTaskWarn
