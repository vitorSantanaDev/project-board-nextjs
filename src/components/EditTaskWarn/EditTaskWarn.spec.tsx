import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import EditTaskWarn from '.'
import { handleClickCancelEdit } from './mockProps'
describe('<EditTaskWarn />', () => {
  it('should render the button', () => {
    render(<EditTaskWarn handleClickCancelEdit={handleClickCancelEdit} />)
    const button = screen.getByRole('button')
    expect(button).toBeInTheDocument()
  })
  it('should call the handleClickCancelEdit function when the button is clicked', () => {
    render(<EditTaskWarn handleClickCancelEdit={handleClickCancelEdit} />)
    const button = screen.getByRole('button')
    userEvent.click(button)
    expect(handleClickCancelEdit).toBeCalledTimes(1)
  })
  it('should render the message "Você está editando uma tarefa!"', () => {
    render(<EditTaskWarn handleClickCancelEdit={handleClickCancelEdit} />)
    const text = screen.getByText('Você está editando uma tarefa!')
    expect(text).toBeInTheDocument()
  })

  it('shoul match snapshot', () => {
    const { container } = render(
      <EditTaskWarn handleClickCancelEdit={handleClickCancelEdit} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
