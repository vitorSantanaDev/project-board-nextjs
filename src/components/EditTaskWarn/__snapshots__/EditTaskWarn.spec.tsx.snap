// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditTaskWarn /> shoul match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 2rem;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
}

.c1 {
  border: none;
  cursor: pointer;
  padding: 0.75rem;
  margin-left: 8px;
  margin-right: 12px;
  background-color: transparent;
}

.c2 {
  color: #ffb800;
  font-size: 1.5rem;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <svg
      color="#FF3636"
      fill="none"
      height="30"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(255, 54, 54);"
      viewBox="0 0 24 24"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
  <span
    class="c2"
  >
    Você está editando uma tarefa!
  </span>
</div>
`;
